src/app/badges/badges.service.ts

```diff
@@ -36,4 +58,30 @@ export class BadgesService {
 
     return playerBadgeDoc.id;
   }
+
+  awardWithBadgeId(playerId, teamId, seasonId, badgeId) {
+    return this.getBadge(badgeId).pipe(
+      flatMap((badge: Badge) => from(this.awardBadge(playerId, teamId, seasonId, badge)))
+    ).subscribe();
+  }
+
+  checkForSpeakerBadges(playerId: string, teamId: string, seasonId: string) {
+    const badges = environment.badges;
+
+    // check if user has competent speaker badge
+    return this.getUserBadge(playerId, badges.competentSpeaker, seasonId).pipe(
+      // award competent speaker badge if it doesn't exist yet
+      tap((badge: PlayerBadge) => !badge ? this.awardWithBadgeId(playerId, teamId, seasonId, badges.competentSpeaker) : null ),
```

Comment:
we need to replace all tap methods here with flatMap to make them return an observable:
```suggestion
      flatMap((badge: PlayerBadge) => badge ? of(badge) : this.awardWithBadgeId(playerId, teamId, seasonId, badges.competentSpeaker)),
      takeWhile((badge: PlayerBadge | number) => typeof badge === 'PlayerBadge')
```